File: src/hash/TwoSum.java
package hash;

import java.util.HashMap;
import java.util.HashSet;

public class TwoSum {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> hashMap = new HashMap<>();
        int[] ans = new int[2];

        for (int i = 0; i < nums.length; i++) {
            if (hashMap.containsKey(target - nums[i])) {
                return new int[]{i, hashMap.get(target - nums[i])};
            } else {
                hashMap.put(nums[i], i);
            }
        }
        return ans;

    }
}

File: LeetcodeKt/src/array/TwoSum.kt
package array

/*
#1. 两数之和
https://leetcode-cn.com/problems/two-sum/
 */
class TwoSum {
    fun twoSum(nums: IntArray, target: Int): IntArray {
        val map = HashMap<Int, Int>()
        for (i in nums.indices) {
            if (map.containsKey(target - nums[i])) {
                return intArrayOf(map.get(target - nums[i])!!, i)
            } else {
                map.put(nums[i], i)
            }
        }
        return intArrayOf(0)
    }
}